php-frontend/src/main/java/org/sonar/php/metrics/MetricsVisitor.java

```diff
@@ -0,0 +1,168 @@
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.Tree.Kind;
+import org.sonar.plugins.php.api.visitors.CheckContext;
+import org.sonar.plugins.php.api.visitors.PHPSubscriptionCheck;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class MetricsVisitor extends PHPSubscriptionCheck {
+
+  private static final Number[] LIMITS_COMPLEXITY_FUNCTIONS = {1, 2, 4, 6, 8, 10, 12, 20, 30};
+  private static final Number[] FILES_DISTRIBUTION_BOTTOM_LIMITS = {0, 5, 10, 20, 30, 60, 90};
+
+  private static final Kind[] FUNCTION_NODES = {
+    Kind.FUNCTION_DECLARATION,
+    Kind.FUNCTION_EXPRESSION,
+    Kind.METHOD_DECLARATION,
+  };
+
+  private final FileSystem fs;
+  private final SensorContext sensorContext;
+  private InputFile inputFile;
+  private NoSonarFilter noSonarFilter;
+  private FileLinesContextFactory fileLinesContextFactory;
+
+  private int classComplexity;
+  private int functionComplexity;
+  private RangeDistributionBuilder functionComplexityDistribution;
+  private RangeDistributionBuilder fileComplexityDistribution;
+
+  public MetricsVisitor(FileSystem fs, SensorContext context, NoSonarFilter noSonarFilter, FileLinesContextFactory fileLinesContextFactory) {
+    this.fs = fs;
+    this.sensorContext = context;
+    this.noSonarFilter = noSonarFilter;
+    this.fileLinesContextFactory = fileLinesContextFactory;
+  }
+
+  @Override
+  public List<Kind> nodesToVisit() {
+    List<Kind> result = new ArrayList<>(Arrays.asList(FUNCTION_NODES));
+    result.add(Kind.CLASS_DECLARATION);
+    result.add(Kind.COMPILATION_UNIT);
+    return result;
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (tree.is(Kind.COMPILATION_UNIT)) {
+      classComplexity = 0;
+      functionComplexity = 0;
+      functionComplexityDistribution = new RangeDistributionBuilder(CoreMetrics.FUNCTION_COMPLEXITY_DISTRIBUTION, LIMITS_COMPLEXITY_FUNCTIONS);
+      fileComplexityDistribution = new RangeDistributionBuilder(CoreMetrics.FILE_COMPLEXITY_DISTRIBUTION, FILES_DISTRIBUTION_BOTTOM_LIMITS);
+      inputFile = fs.inputFile(fs.predicates().is(context().file()));
+
+    } else if (tree.is(Kind.CLASS_DECLARATION)) {
+      classComplexity += NewComplexityVisitor.complexity(tree);
+
+    } else if (tree.is(FUNCTION_NODES)) {
+      int currentFunctionComplexity = NewComplexityVisitor.complexityWithoutNestedFunctions(tree);
+      functionComplexity += currentFunctionComplexity;
+      functionComplexityDistribution.add(currentFunctionComplexity);
+    }
+  }
+
+  @Override
+  public void leaveNode(Tree tree) {
+    if (tree.is(Kind.COMPILATION_UNIT)) {
+      saveComplexityMetrics(context());
+      saveCounterMetrics(context());
+      saveLineMetrics(context());
+    }
+  }
+
+  private void saveCounterMetrics(CheckContext context) {
+// FIXME UNCOMMENT TO ENABLE COUNTER VISITOR
+//    CounterVisitor counter = new CounterVisitor(context.tree());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)